build-system/babel-plugins/babel-plugin-transform-inline-isenumvalue/index.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {dirname, join: joinPath} = require('path');
+const {readFileSync} = require('fs');
+
+/**
+ * @fileoverview
+ * Takes isEnumValue() calls and replaces them with a smaller form that only
+ * requires the enum object's values, and not its property keys.
+ *
+ * In:
+ *     isEnumValue({FOO: 'foo', BAR: 'bar'}, x);
+ * Out:
+ *     x === 'foo' || x === 'bar'
+ */
+
+module.exports = function (babel) {
+  const {types: t, parseSync, traverse} = babel;
+
+  function resolveImportEvaluate(path, filename) {
+    const evaluated = path.evaluate();
+    if (evaluated.confident) {
+      return evaluated;
+    }
+    if (path.isIdentifier()) {
+      const importDeclaration = path.scope.getBinding(path.node.name).path
+        .parentPath;
+      if (importDeclaration.isImportDeclaration()) {
+        const importPath = joinPath(
+          dirname(filename),
+          importDeclaration.node.source.value
+        );
+        let code;
+        for (const suffix of ['.js', '/index.js', '.mjs', '/index.mjs']) {
+          try {
+            code = readFileSync(importPath + suffix, 'utf8');
+            break;
+          } catch (_) {}
+        }
+        if (code) {
+          const specifier = importDeclaration.node.specifiers.find(
+            ({local}) => local.name === path.node.name
+          );
+          let evaluated;
+          traverse(parseSync(code), {
+            Identifier(path) {
```

Comment:
You can probably use `Program` here, to avoid a full walk of the file. We know it's an export somehwere:

```js
Program(path) {
  path.stop(); // prevent a full walk
  path.scope.getBinding(name).path.evaluate()
}
```

Reply:
nice!